WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewModel.swift

```diff
@@ -45,6 +45,11 @@ final class ProductFormViewModel: ProductFormViewModelProtocol {
         originalProduct
     }
 
+    /// Whether the "Promote with Blaze" button should show Blaze intro view first or not when tapped.
+    var shouldShowBlazeIntroView: Bool {
+        blazeCampaignResultsController.isEmpty
```

Comment:
FYI: you can also call `storageManager.viewStorage.loadAllBlazeCampaigns(siteID: product.siteID).isEmpty` here without the need for the results controller. But since we want to observe the storage to hide the Promote button later, it makes more sense to keep the results controller.